### PR TITLE
[SMTChecker] Test that non-Boolean literals are actually integers

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

File: libsolidity/formal/CVC4Interface.cpp [hunk 1]
```diff
@@ -132,8 +132,17 @@ CVC4::Expr CVC4Interface::toCVC4Expr(Expression const& _expr)
 		else if (n == "false")
 			return m_context.mkConst(false);
 		else
-			// We assume it is an integer...
+		{
```
Comment (ekpyron) [inline libsolidity/formal/CVC4Interface.cpp]: You could get rid of these parenthesis, if you move the ``return`` into the ``try`` block...

File: libsolidity/formal/CVC4Interface.cpp [hunk 2]
```diff
@@ -132,8 +132,17 @@ CVC4::Expr CVC4Interface::toCVC4Expr(Expression const& _expr)
...
+			try
+			{
+				bigint isNumber(n);
+			}
+			catch (...)
```
Comment (ekpyron) [inline libsolidity/formal/CVC4Interface.cpp:139]: Any reason for not just letting the exception be thrown directly? ``solAssert(false, "");`` will only throw another one anyways... not sure about it. Generally I'm still not sure whether this is the nicest way of checking for numbers... And: doesn't ``CVC4::Rational(n)`` throw already if it fails to parse? Looks like it from peeking at the source and looking at the underlying GMP docs...

**Comment by leonardoalt** [comment]
@ekpyron updated

**Review by ekpyron — APPROVED** [review]
One could even catch ``std::exception const& _e`` and use ``string("Z3/CVC4 could not convert literal to integer: ") + _e.what()`` as message for ``solAssert``... but since ``solAssert`` will print file name and line numbers and this shouldn't be happening anyways, I'd say it's fine as it is!

**Comment by leonardoalt** [comment]
Yea I almost never use messages on solAssert, you know where it crashed anyway

**Pull request merged**